src/pydrex/io.py

```diff
@@ -0,0 +1,141 @@
+
+import pydrex.exceptions as _err
+
+
+def parse_params(file):
+    """Parse an INI file containing PyDRex parameters."""
+    config = configparser.ConfigParser()
+    configpath = _resolve_path(file)
+    config.read(configpath)
+
+    geometry = config["Geometry"]
+    output = config["Output"]
+    params = config["Parameters"]
+
+    mesh = read_mesh(_resolve_path(geometry.get("meshfile"), configpath.parent))
+    olivine_fraction = params.getfloat("olivine_fraction")
+    enstatite_fraction = params.getfloat("enstatite_fraction", 1.0 - olivine_fraction)
+
+    # TODO: Allow user-given order for Euler angle output (like Fraters 2021)?
+    olivine_output = tuple(
+        output.get("olivine", fallback="volume_fraction, rotation_matrix")
+        .replace(" ", "")
+        .split(",")
+    )
+    enstatite_output = tuple(
+        output.get("enstatite", fallback="volume_fraction, rotation_matrix")
+        .replace(" ", "")
+        .split(",")
+    )
+
+    return dict(
+        zip(
+            [
+                "mesh",
+                "simulation_name",
+                "checkpoint_interval",
+                "olivine_output",
+                "enstatite_output",
+                "olivine_fraction",
+                "enstatite_fraction",
+                "stress_exponent",
+                "gbm_mobility",
+                "gbs_threshold",
+                "nucleation_efficiency",
+                "olivine_fabric",
+                "number_of_grains",
+            ],
+            [
+                mesh,
+                output.get("simulation_name", fallback="simulation_name"),
+                output.getint(
+                    "checkpoint_interval", fallback=0
+                ),  # 0 disables checkpointing
+                ["olivine_" + s for s in olivine_output],  # Unique names for np.savez()
+                [
+                    "enstatite_" + s for s in enstatite_output
+                ],  # Unique names for np.savez()
+                olivine_fraction,
+                enstatite_fraction,
+                params.getfloat("stress_exponent"),
+                params.getfloat("gbm_mobility"),
+                params.getfloat("gbs_threshold"),
+                params.getfloat("nucleation_efficiency"),
+                params.get("olivine_fabric"),
+                params.getint("number_of_grains"),
+            ],
+        )
+    )
+
+
+def _resolve_path(path, refdir=None):
+    cwd = pathlib.Path.cwd()
+    if refdir is None:
+        _path = cwd / path
+    else:
+        _path = cwd / refdir / path
+    if _path.is_file():
+        return _path.resolve()
+    raise IOError(f"file '{_path}' does not exist")
+
+
+def read_mesh(meshfile):
+    """Read mesh from `meshfile` into numpy arrays.
+
+    Supported formats:
+    - Python script that assigns grid coordinates (2D Numpy array, see `create_mesh`)
+
+    """
+    if meshfile.suffix == ".py":
+        return create_mesh(runpy.run_path(meshfile)["gridcoords"])
```

Comment:
I am wondering if `importlib` should be used instead of `runpy`, but I am probably worrying for no reason.

Reply:
I think for importlib it has to be a proper module, with an `__init__.py` or something:

```
>>> import importlib as i
>>> i.import_module("data/2d_mesh_test.py")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    i.import_module("data/2d_mesh_test.py")
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'data/2d_mesh_test'
```

Reply:
I agree that having the "mesh file" be executable is not ideal, it would be best to read from an established mesh format like GMSH (or maybe something more stable...)

Reply:
Something like that works, but it is far from a one-liner! Haha
```
import importlib
import sys

spec = importlib.util.spec_from_file_location('my_script_name',
                                              'path_to_my_script')
my_script = importlib.util.module_from_spec(spec)
sys.modules['my_script_name'] = my_script
spec.loader.exec_module(my_script)

print(my_script.my_variable)
```

Reply:
I'm sticking with `runpy` here!